lib/core/HeadRequestAsyncTask.js

```diff
@@ -34,6 +35,21 @@ var _execute = function (callback) {
 		self.callback(null, result);
 	};
 
+    // use http proxy if configured
+    // source from: http://stackoverflow.com/questions/3862813/how-can-i-use-an-http-proxy-with-node-js-http-client
+    var match;
+    if (process.env.http_proxy != null) {
+        match = process.env.http_proxy.match(/^(http:\/\/)?([^:\/]+)(:([0-9]+))?/i);
```

Comment:
This is a good default, as this is the "usual" name that other browsers and
download agents look for. The name (and whether it is even looked for)
could be made a config option. This simple approach does not try to deal
with more advanced proxy config, e.g. https_proxy and no_proxy settings.
..Dave

On Tue, Sep 22, 2015 at 4:35 AM, Tushar Mathur <notifications@github.com>
wrote:

> In lib/core/HeadRequestAsyncTask.js
> <https://github.com/tusharmath/Multi-threaded-downloader/pull/44#discussion_r40071982>
> :
>
> > @@ -34,6 +35,21 @@ var _execute = function (callback) {
> >  		self.callback(null, result);
> >  	};
> >
> > +    // use http proxy if configured
> > +    // source from: http://stackoverflow.com/questions/3862813/how-can-i-use-an-http-proxy-with-node-js-http-client
> > +    var match;
> > +    if (process.env.http_proxy != null) {
> > +        match = process.env.http_proxy.match(/^(http:\/\/)?([^:\/]+)(:([0-9]+))?/i);
>
> What if the env variable name is something else?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tusharmath/Multi-threaded-downloader/pull/44/files#r40071982>
> .
>